xbmc/addons/addoninfo/AddonType.cpp

```diff
@@ -24,6 +24,14 @@ void CAddonType::SetProvides(const std::string& content)
 {
   if (!content.empty())
   {
+    /*
+     * Normally the "provides" becomes added from xml scan, but from database can
```

Comment:
...but for add-ons stored in the database (e.g. repository contents) it might not be available. Since this information is available in add-on metadata for the main type (see extrainfo) we take the function contents and insert it if empty.